### PR TITLE
Ensure custom command line args and env vars are used in test actions

### PR DESCRIPTION
**Pull request by cgrindel**
Related to #906.

- Fixed logic for all schemes to ensure that `buildMode` specific environment variables (e.g., `bazel`) are included in launch and test actions.
- Add sort and merge operations for `XCScheme.EnvironmentVariable` `Sequence` values. It applies a precedence for overriding environment variables and ensures that the variable lists are consistent for equality and diff purposes.

### CODE REVIEW

File: test/fixtures/generator/bwx.xcodeproj/xcshareddata/xcschemes/generator.xcscheme [hunk 1]
```diff
@@ -91,7 +91,7 @@
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       customLLDBInitFile = "$(BAZEL_LLDB_INIT)"
-      shouldUseLaunchSchemeArgsEnv = "YES">
+      shouldUseLaunchSchemeArgsEnv = "NO">
```
Comment (brentleyjones) [inline test/fixtures/generator/bwx.xcodeproj/xcshareddata/xcschemes/generator.xcscheme:94]: What is the affect of this? Does this mean the `BAZEL_WORKSPACE_DIRECTORY` env isn't passed anymore? If so, we need that set explicitly for https://github.com/bazelbuild/apple_support/pull/128 to work.

Comment (cgrindel) [reply test/fixtures/generator/bwx.xcodeproj/xcshareddata/xcschemes/generator.xcscheme:94]: This unchecks the box in the Xcode dialog allowing the custom args/env vars to be used.

I will update the code to ensure that the bazel env vars are included.

Comment (cgrindel) [reply test/fixtures/generator/bwx.xcodeproj/xcshareddata/xcschemes/generator.xcscheme:94]: If the magic Bazel env vars need to be set in every BwB scheme, then the current logic for custom schemes is not quite right. It only includes them if the a launch action is defined. 

@brentleyjones Can you confirm that we do want the magic Bazel env vars set in the launch action for every BwB scheme?

Comment (brentleyjones) [reply test/fixtures/generator/bwx.xcodeproj/xcshareddata/xcschemes/generator.xcscheme:94]: We want them set in both the launch and test actions, yes. While Bazel doesn't set `BAZEL_WORKSPACE_DIRECTORY` by default for tests, we still need it for the above feature I liked.

Comment (brentleyjones) [reply test/fixtures/generator/bwx.xcodeproj/xcshareddata/xcschemes/generator.xcscheme:94]: Looks like this one doesn't have the bazel magic env variables still?

Comment (cgrindel) [reply test/fixtures/generator/bwx.xcodeproj/xcshareddata/xcschemes/generator.xcscheme:94]: This file is for BwX. I am seeing the magic env vars in the test action in the BwB version of the scheme.

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**